examples/ensemble/plot_clustering_on_random_trees_embedding_outputs_with_modified_distance_metrics.py

```diff
@@ -0,0 +1,177 @@
+"""
+=========================================================================
+Clustering on RandomTreesEmbedding Outputs with Modified Distance Metrics
+=========================================================================
+This demonstration is inspired by the Sparse Projection Oblique
+Randomer Forest(SPORF) package. 
+
+Distance is an important concept in clustering and other machine
+learning tasks that gives us an intuitive sense of the similarity
+between samples. This example shows how can we generate different 
+distance metrics from the tree structure info of 
+Random Tree Embedding and use these distance metrics to complete 
+various tasks. In this example, we will use them to do clustering. 
+
+The distances between two samples i and j can be defined as: 
+1) the depth of the nearest common ancestor of the leaf nodes
+sample i and j land into (take the reciprocal to make it a 
+distance); 
+2) the length of the shortest path of the leaf nodes
+sample i and j land into; 
+3) the probability of sample i and j landing into the same
+lead node(take 1 minus the value to make it a distance). 
+
+The example will be implemented on the Iris dataset as a
+simple demonstration. And the performance will be evaluated
+using ARI. 
+
+"""
+
+from sklearn import tree
+from sklearn import datasets
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.ensemble import RandomTreesEmbedding
+
+# use the Iris dataset as a simple example
+
+iris = datasets.load_iris()
+
+samples = iris.data
+labels = iris.target
+n_sample = samples.shape[0]
+n_feature = samples.shape[1]
+n_estimator = 200
+
+plt.figure(figsize=(15, 10))
+
+# train a random trees embedding model of 200 estimators
+
+clf = RandomTreesEmbedding(n_estimators=n_estimator, max_depth = 10)
+clf = clf.fit(samples)
+
+# compute the first distance metrics, which is 
+# one over the depth of the nearest common ancestor
+# averaged over all estimators
+
+forest = clf.estimators_
+d = np.zeros([n_sample, n_sample])
+
+for k in range(n_estimator):
```

Comment:
You should write functions for each of the three methods you are comparing, probably makes sense for them to take in an estimator and output a distance matrix or something like that

Reply:
This will just be more clear to the reader

Reply:
Makes sense. Changed accordingly.